File: content/snippets/python/exercises/beginner/max_of_array.py
import sys

def max_of_array(arr):
    return max(arr) if arr else None

if __name__ == '__main__':
    try:
        parts = sys.stdin.read().strip().split()
        if parts:
            n = int(parts[0]); vals = list(map(int, parts[1:1+n]))
        else:
            n = 5; vals = [1,2,3,4,5]
    except Exception:
        n = 5; vals = [1,2,3,4,5]
    print(max_of_array(vals))

File: content/snippets/typescript/exercises/beginner/max_of_array.ts
const fs = require('fs');
const input = fs.readFileSync(0, 'utf8').trim().split(/\s+/);
let idx = 0;
let n = input.length ? Number(input[idx++]) : 5;
const arr = [];
for (let i = 0; i < n; i++) arr.push(input.length > idx ? Number(input[idx++]) : i+1);
console.log(Math.max(...arr));
